src/metabase/plugins.clj

```diff
@@ -17,41 +18,48 @@
   (or (env/env :mb-plugins-dir)
       (.getAbsolutePath (io/file "plugins"))))
 
-;; logic for determining plugins dir -- see below
-(defonce ^:private plugins-dir*
-  (delay
-    (let [filename (plugins-dir-filename)]
-      (try
-        ;; attempt to create <current-dir>/plugins if it doesn't already exist. Check that the directory is readable.
-        (let [path (u.files/get-path filename)]
-          (u.files/create-dir-if-not-exists! path)
-          (assert (Files/isWritable path)
-            (trs "Metabase does not have permissions to write to plugins directory {0}" filename))
-          path)
-        ;; If we couldn't create the directory, or the directory is not writable, fall back to a temporary directory
-        ;; rather than failing to launch entirely. Log instructions for what should be done to fix the problem.
-        (catch Throwable e
-          (log/warn
-           e
-           (trs "Metabase cannot use the plugins directory {0}" filename)
-           "\n"
-           (trs "Please make sure the directory exists and that Metabase has permission to write to it.")
-           (trs "You can change the directory Metabase uses for modules by setting the environment variable MB_PLUGINS_DIR.")
-           (trs "Falling back to a temporary directory for now."))
-          ;; Check whether the fallback temporary directory is writable. If it's not, there's no way for us to
-          ;; gracefully proceed here. Throw an Exception detailing the critical issues.
-          (let [path (u.files/get-path (System/getProperty "java.io.tmpdir"))]
-            (assert (Files/isWritable path)
-              (trs "Metabase cannot write to temporary directory. Please set MB_PLUGINS_DIR to a writable directory and restart Metabase."))
-            path))))))
-
-;; Actual logic is wrapped in a delay rather than a normal function so we don't log the error messages more than once
-;; in cases where we have to fall back to the system temporary directory
-(defn- plugins-dir
-  "Get a `Path` to the Metabase plugins directory, creating it if needed. If it cannot be created for one reason or
-  another, or if we do not have write permissions for it, use a temporary directory instead."
+(def ^:private plugins-dir*
```

Comment:
Not that it makes any difference to running things but I think it's nice to add `^{:arglists '([filename])}` just so Kondo knows it's a function and can do arg-count checks and so it pops up in your editor auto-complete